Plutarch/Internal.hs

```diff
@@ -104,6 +104,15 @@ mapTerm f (TermResult t d) = TermResult (f t) d
 
 mkTermRes :: RawTerm -> TermResult
 mkTermRes r = TermResult r []
+
+-- | Type of `s`.
+data S
```

Comment:
It maybe worth documenting this `S`. Why do we have a `s` in `Term s`?

Reply:
We could simply point to the part of the users guide explaining the `s`